internal/devices/drivers.go

```diff
@@ -12,19 +12,23 @@ import (
 	"github.com/Microsoft/hcsshim/internal/log"
 	"github.com/Microsoft/hcsshim/internal/resources"
 	"github.com/Microsoft/hcsshim/internal/uvm"
-	"github.com/pkg/errors"
+	"github.com/google/uuid"
 )
 
-// InstallKernelDriver mounts a specified kernel driver, then installs it in the UVM.
+// InstallDriver mounts a share from the host into the UVM, installs any kernel drivers in the share,
+// and configures the environment for library files and/or binaries in the share.
 //
-// `driver` is a directory path on the host that contains driver files for standard installation.
+// InstallDriver mounts a specified kernel driver, then installs it in the UVM.
+//
+// `share` is a directory path on the host that contains files for standard driver installation.
 // For windows this means files for pnp installation (.inf, .cat, .sys, .cert files).
 // For linux this means a vhd file that contains the drivers under /lib/modules/`uname -r` for use
-// with depmod and modprobe.
+// with depmod and modprobe. For GPU, this vhd may also contain library files (under /usr/lib) and
+// binaries (under /usr/bin or /usr/sbin) to be used in conjunction with the modules.
 //
 // Returns a ResourceCloser for the added mount. On failure, the mounted share will be released,
 // the returned ResourceCloser will be nil, and an error will be returned.
-func InstallKernelDriver(ctx context.Context, vm *uvm.UtilityVM, driver string) (closer resources.ResourceCloser, err error) {
+func InstallDrivers(ctx context.Context, vm *uvm.UtilityVM, share string, gpuDriver bool) (closer resources.ResourceCloser, err error) {
```

Comment:
does this function really need to know about GPU drivers? Maybe we can just pass the read/write path as a parameter? either always explicitly or only generate a GUID when it's empty?

Reply:
I thought about that, but since we don't need that value at all for windows guests, it felt easier to have a single bool instead of moving logic around. What do you think?